### ISSUE TITLE
Mixer crash loops until Galley is available

### ISSUE THREAD

**Issue by Stono** [istio/istio#15098]
**Bug description**
Enabling Galley and MCP results in Mixer components crash looping until Galley is fully available and able to handle requests.

```
2019-06-24T17:39:06.387193Z     info    Secure MCP configured. Waiting for required certificate files to become available: [/etc/certs/cert-chain.pem /etc/certs/key.pem /etc/certs/root-cert.pem]
2019-06-24T17:39:06.387227Z     info    /etc/certs/cert-chain.pem found
2019-06-24T17:39:06.387265Z     info    /etc/certs/key.pem found
2019-06-24T17:39:06.387277Z     info    /etc/certs/root-cert.pem found
2019-06-24T17:39:06.388951Z     info    parsed scheme: ""
2019-06-24T17:39:06.389080Z     info    scheme "" not registered, fallback to default scheme
2019-06-24T17:39:06.389171Z     info    ccResolverWrapper: sending update to cc: {[{istio-galley.istio-system.svc:9901 0  <nil>}] }
2019-06-24T17:39:06.389268Z     info    ClientConn switching balancer to "pick_first"
2019-06-24T17:39:06.389537Z     info    pickfirstBalancer: HandleSubConnStateChange: 0xc4204f6010, CONNECTING
2019-06-24T17:39:06.389566Z     info    Awaiting for config store sync...
2019-06-24T17:39:06.389623Z     info    mcp     (re)trying to establish new MCP sink stream
```

It's actually the livenessProbe doing the killing:
```
8s          Normal    Killing             Pod                   Killing container with id docker://mixer:Container failed liveness probe.. Container will be killed and recreated.
```

Which is configured quite aggressively (this is telemetry):
```
    livenessProbe:
      failureThreshold: 3
      httpGet:
        path: /version
        port: 15014
        scheme: HTTP
      initialDelaySeconds: 5
      periodSeconds: 5
      successThreshold: 1
      timeoutSeconds: 1
```

This only happens on a fresh start of mixer, if you scale galley to 0 **after** an initial successful connection, it stays up.

**Affected product area (please put an X in all that apply)**

[ ] Configuration Infrastructure
[ ] Docs
[x] Installation
[ ] Networking
[ ] Performance and Scalability
[x] Policies and Telemetry
[ ] Security
[ ] Test and Release
[x] User Experience
[ ] Developer Infrastrcture

**Expected behavior**
Mixer shouldn't crash on initial start up failing to get to galley, it should just perhaps be "unready" until it's made its initial connection and got what it needs in order to start servicing requests.

**Steps to reproduce the bug**
scale galley to 0, restart mixer

**Version (include the output of `istioctl version --remote` and `kubectl version`)**
Istio 1.2.0

**How was Istio installed?**
Helm

**Environment where bug was observed (cloud vendor, OS, etc)**
GKE

**Comment by douglas-reid** [comment]
This has been discussed in a number of forums previously. I feel like there was even a PR to address this open at one point, but I'm struggling to find that PR now.

Related (somewhat):
- https://github.com/istio/istio/issues/12863
- https://github.com/istio/istio/issues/12226

**Comment by Stono** [comment]
It's totally fine, we run without galley once again, it was fixed in 1.2.1
